src/mscorlib/src/System/Text/StringBuilder.cs

```diff
@@ -561,6 +561,14 @@ void ISerializable.GetObjectData(SerializationInfo info, StreamingContext contex
             if (repeatCount==0) {
                 return this;
             }
+
+            // this is where we can check if the repeatCount will put us over m_MaxCapacity
+            // We are doing the check here to prevent the corruption of the StringBuilder.
+            int newLength = Length + repeatCount;
+            if (newLength > m_MaxCapacity || newLength < repeatCount) {
+                throw new ArgumentOutOfRangeException("repeatCount", Environment.GetResourceString("ArgumentOutOfRange_NegativeCount"));
```

Comment:
I wanted to keep the same error message in case people are checking the message. A different one could be.
```
ArgumentOutOfRange_LengthGreaterThanCapacity = The length cannot be greater than the capacity.
```

If we are not worried about not changing the message in the exception, then I can update it to use the better message.